### PR TITLE
Fix concat feature approach in CNN

### LINKED ISSUE #314: Feature approach concat does not work
Currently this approach has been implemented with `tf.concat` which expects a list of tensors to concatenate. Instead CNN constructs a matrix of attention_heads, embeddeding_dimension so the concat essentially needs to just flatten that into one vector

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

As I was trying to experiment with multiple attention heads and concatenating them to increase the representational power of the network I released that the implementation was not working as expected. 

Fixes #314 

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v1
         with:
-          python-version: 3.7.10
+          python-version: 3.7.11
```
Comment (aCampello) [inline .github/workflows/main.yml:22]: rebase

File: wellcomeml/ml/cnn.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def residual_attention(x1):
         elif self.feature_approach == 'sum':
             x = tf.keras.backend.sum(x, axis=1)
         elif self.feature_approach == 'concat':
-            x = tf.concat(x, axis=1)
+            x = tf.keras.layers.Flatten()(x)
```
Comment (aCampello) [inline wellcomeml/ml/cnn.py:227]: So you wanted one big one-dimensional vector and that one was only squashing one dimension?

**Review by aCampello — APPROVED** [review]

File: wellcomeml/ml/cnn.py (hunk 2 above)
Comment (nsorros) [reply wellcomeml/ml/cnn.py:227]: the one there was trying to concatenate two matrix on a particular axis instead of squashing so for example if you had 2 matrices of `(32,25,400)` it would create `(32,50,400)` whereas I only had one matrix to begin with and I wanted a matrix `(32,10_000)`

File: .github/workflows/main.yml (hunk 1 above)
Comment (nsorros) [reply .github/workflows/main.yml]: done
